ecosystem/indexer-grpc/indexer-grpc-utils/src/file_store_operator/gcs.rs

```diff
@@ -155,27 +162,40 @@ impl FileStoreOperator for GcsFileStoreOperator {
     async fn upload_transaction_batch(
         &mut self,
         _chain_id: u64,
-        transactions: Vec<EncodedTransactionWithVersion>,
+        transactions: Vec<Transaction>,
     ) -> anyhow::Result<(u64, u64)> {
-        let start_version = transactions.first().unwrap().1;
-        let end_version = transactions.last().unwrap().1;
+        let start_version = transactions.first().unwrap().version;
+        let end_version = transactions.last().unwrap().version;
         let batch_size = transactions.len();
         anyhow::ensure!(
-            start_version % BLOB_STORAGE_SIZE as u64 == 0,
+            start_version % FILE_ENTRY_TRANSACTION_COUNT == 0,
             "Starting version has to be a multiple of BLOB_STORAGE_SIZE."
         );
         anyhow::ensure!(
-            batch_size % BLOB_STORAGE_SIZE == 0,
+            batch_size % FILE_ENTRY_TRANSACTION_COUNT as usize == 0,
             "The number of transactions to upload has to be multiplier of BLOB_STORAGE_SIZE."
         );
-
+        let start_time = std::time::Instant::now();
         let bucket_name = self.bucket_name.clone();
-        let current_batch = transactions.iter().cloned().collect_vec();
-        let transactions_file = build_transactions_file(current_batch).unwrap();
+        let file_entry: FileEntry =
+            FileEntryBuilder::new(transactions, self.storage_format).try_into()?;
+        let file_entry_key = FileEntryKey::new(start_version, self.storage_format).to_string();
+        log_grpc_step(
```

Comment:
this is redundant, this step is already logged in the filestore worker. we do need to log the filestore compression step though.

Reply:
done.